File: 2.Dagger/Basics/InjectionTypes/src/main/java/com/app/basics/MainActivity.java
package com.app.basics;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {
    private AppComponent component;

    @Inject
    String providedString;

    @Inject
    SimpleClass simpleClass;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        component = DaggerAppComponent.builder().build();
        component.inject(this);

        Log.i(TAG, "property injection in Activity: " + providedString);

        Log.i(TAG, "constructor injection in Activity: " + simpleClass.toString());
    }

    //При иньекции в метод - Dagger сам вызовет данный метод.
    //Говорят это может пригодиться при инициализации параметров класс после создания его экземпляра.
    //Dagger доолжен понимать где взять каждый параметр метода. Если добавим один неизвестный, то он не сбилдится.

    @Inject
    public void testMethod(String providedString){
        Log.i(TAG, "property injection in Activity method:" + providedString);
    }

    private static final String TAG = "MainActivity";
}

File: 3.Koin/Basics/FirstExample/src/main/java/com/app/basics/MainActivity.kt
package com.app.basics

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import org.koin.android.ext.android.inject

class MainActivity : AppCompatActivity() {

    private val str: String by inject()
    private val a:A by inject()
    private val value: Boolean by inject()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Log.i(TAG, "onCreate: $str")
        Log.i(TAG, "onCreate: ${a.number}")

        val button = findViewById<Button>(R.id.button)
        button.setOnClickListener {
            Log.i(TAG, "button $value")
        }
    }

    companion object {
        private const val TAG = "MainActivity"
    }
}